pkg/publicapi/server.go

```diff
@@ -211,27 +212,54 @@ func (apiServer *Server) ListenAndServe(ctx context.Context) error {
 		}
 	}
 
-	log.Ctx(ctx).Debug().Msgf(
-		"API server listening for host %s on %s...", apiServer.Address, listener.Addr().String())
-
+	// Create an error channel
+	errChan := make(chan error, 1)
 	go func() {
-		var err error
+		defer close(errChan)
+		var serveErr error
 
 		if apiServer.useTLS {
-			err = apiServer.httpServer.ServeTLS(listener, apiServer.TLSCertificateFile, apiServer.TLSKeyFile)
+			serveErr = apiServer.httpServer.ServeTLS(listener, apiServer.TLSCertificateFile, apiServer.TLSKeyFile)
+			if serveErr != nil {
+				serveErr = fmt.Errorf(
+					"failed to start API server with TLS (cert path: %s) (key path: %s): %w",
+					apiServer.TLSCertificateFile,
+					apiServer.TLSKeyFile,
+					serveErr,
+				)
+			}
 		} else {
-			err = apiServer.httpServer.Serve(listener)
+			serveErr = apiServer.httpServer.Serve(listener)
+			if serveErr != nil {
+				serveErr = fmt.Errorf(
+					"failed to start API server: %w", err)
+			}
 		}
 
-		if err == http.ErrServerClosed {
-			log.Ctx(ctx).Debug().Msgf(
+		if serveErr == http.ErrServerClosed {
+			log.Ctx(ctx).Info().Msgf(
 				"API server closed for host %s on %s.", apiServer.Address, apiServer.httpServer.Addr)
-		} else if err != nil {
-			log.Ctx(ctx).Err(err).Msg("Api server can't run. Cannot serve client requests!")
+		} else if serveErr != nil {
+			log.Ctx(ctx).Error().Err(err).Msg("Api server can't run. Cannot serve client requests")
+			errChan <- serveErr
 		}
 	}()
 
-	return nil
+	// NB(forrest): There are few cases when http.Serve* will return an error,
+	// one of which is when there are issues reading the TLS configuration.
+	// In most cases, if the server starts successfully, http.Serve* will block and not return an error.
+	// We use a select statement to wait for either an error from errChan or a timeout of 2 seconds.
+	// If an error is received within the timeout, it means the server encountered an issue during startup.
+	// If no error is received within the timeout, we assume the server started successfully.
+	// TODO: call os.Exit(1) or log.Fatal if the server stops running for a reason other than ErrServerClosed
+	select {
+	case startErr := <-errChan:
+		return startErr
+	case <-time.After(time.Second * 2):
```

Comment:
It seems a shame to introduce a 2 second pause to startup just for the log message.
How about logging this message at the start of the goroutine instead?
Then we can replace the time.After with a ctx.Done() check

Reply:
Yeah, on second look I don't think this is the right change. I wrote it as when deploying bacalhau with terraform. The user running bacalhau didn't have permissions to read the TLS config. Definitely a rare case, but annoying to debug since the Bacalhau process kept running but didn't respond to any requests. Further I failed to see `failed to start API server` in the log messages.

How would we feel about calling `log.Fatal` in the event the server returns an error? If we did that the user would see:
`15:03:12.041 | FTL pkg/publicapi/server.go:231 > Api server can't run. Cannot serve client requests! error="failed to start server" [NodeID:n-fde9a353]` and the process would exit. I suggest this as I don't see any value in running bacalhau if the API server isn't active. On the other hand this might leave some datastores in a weird state.

Alternatively I am happy to just close this PR and discount my experience as a one-off issue that probably won't happen again.

Reply:
> How would we feel about calling log.Fatal in the event the server returns an error

I am 100% in favour of it visibily crashing rather than quietly failing :D